prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -182,14 +210,29 @@ def immediate_charging(
     ]
     trips["charging_allowed"] = trips[allowed_cols].apply(all, axis=1)
 
+    trips["dwell_charging"] = (
+        trips["charging_allowed"] * trips["dwell_time"] * power * charging_efficiency
+    )
+
+    grouped_trips = trips.groupby("vehicle_number")
+    for vehicle_num, group in grouped_trips:
+        trips.loc[group.index, "max_charging"] = trips.loc[
+            group.index, "dwell_charging"
+        ].sum()
+        trips.loc[group.index, "required_charging"] = (
+            trips.loc[group.index, "trip_miles"].sum() * kwhmi
+        )
```

Comment:
The `kwhmi` multiplier would need to be included (suggested edit below, but @BainanXia, please confirm that this would work).  As far as their equivalence, it looks reasonable to me, but it's also taking Python code to a level I'm still learning.
```
trips["max_charging"] = trips["dwell_charging"].groupby("vehicle_number").transform('sum')
trips["required_charging"] = trips["trip_miles"].groupby("vehicle_number").transform('sum')*kwhmi
```

Reply:
Good catch. I think it should work with the multiplier as well.